src/ua.c

```diff
@@ -932,6 +934,192 @@ int ua_uri_complete(struct ua *ua, struct mbuf *buf, const char *uri)
 }
 
 
+static bool uri_only_user(const struct uri *uri)
+{
+	bool ret;
+	struct sa sa;
+
+	/* Note:
+	 * If only user is given then uri_decode sets uri->host instead of
+	 * uri->user. We don't know if this is a bug. But if somebody changes
+	 * this behavior then the following line has to be adapted.          */
+	ret = pl_isset(&uri->host) && !pl_isset(&uri->user);
+
+	/* exclude IP addresses */
+	ret = ret && 0 != sa_set(&sa, &uri->host, 0);
+
+	return ret;
+}
+
+
+static bool uri_user_and_host(const struct uri *uri)
+{
+	bool ret;
+
+	ret = pl_isset(&uri->host) && pl_isset(&uri->user);
+
+	return ret;
+}
+
+
+static bool uri_host_local(const struct uri *uri)
+{
+	const struct sa *sal;
+	struct sa sap;
+	int err;
+
+	if (!uri)
+		return false;
+
+	if (!pl_strcmp(&uri->host, "localhost"))
+		return true;
+
+	if (!pl_strcmp(&uri->host, "127.0.0.1"))
+		return true;
+
+	if (!pl_strcmp(&uri->host, "::1"))
+		return true;
+
+	sal = net_laddr_af(baresip_network(), AF_INET);
+
+	err = sa_set(&sap, &uri->host, 0);
+	if (err)
+		return false;
+
+	if (sal && sa_cmp(sal, &sap, SA_ADDR))
+		return true;
+
+#ifdef HAVE_INET6
+	sal = net_laddr_af(baresip_network(), AF_INET6);
+	if (sal && sa_cmp(sal, &sap, SA_ADDR))
+		return true;
+#endif
+
+	return false;
+}
+
+
+static bool acc_inactive(const struct account *acc)
+{
+	const struct uri *uri;
+
+	if (acc->regint)
+		return false;
+
+	uri = &acc->luri;
+	if (!uri_host_local(uri))
+		return true;
+
+	return false;
+}
+
+
+static bool uri_mismatch_af(
+		const struct uri *accu, const struct uri *peeru);
+static bool match_port(uint16_t port1, uint16_t port2, bool tls);
+static bool uri_match_transport(const struct uri *accu,
+		const struct uri *peeru, enum sip_transp tp);
+
+
+/**
+ * Find a User-Agent (UA) best fitting for an INVITE to given SIP uri
+ *
+ * @param req_uri The SIP uri for the INVITE
+ *
+ * @return User-Agent (UA) if found, otherwise NULL
+ */
+static struct ua *uag_find_connect(const struct pl *req_uri)
```

Comment:
I think this function should be global, so that it can be used
by any module/app.